src/createLambdaExtensionClient.ts

```diff
@@ -0,0 +1,100 @@
+import type { Context } from 'aws-lambda';
+import { datadog, sendDistributionMetric } from 'datadog-lambda-js';
+
+import { LambdaExtensionMetricsClient } from './LambdaExtensionMetricsClient';
+
+interface DatadogMetric {
+  name: string;
+  tags?: string[];
+  value: number;
+}
+
+interface DatadogConfig {
+  name: string;
+  metrics: boolean;
+}
+
+type Handler<Event, Output> = (
+  event: Event,
+  ctx: Readonly<Context>,
+) => Promise<Output>;
```

Comment:
IIRC there will be an issue with this `Context`: importing it on L1 means that `tsc` will fail to type check the package in `node_modules` unless the consumer also installs `@types/aws-lambda`. This will be the case even if the consumer does not deploy an AWS Lambda.

I can see two ways forward:

1. We move `@types/aws-lambda` to `dependencies`, which is relatively harmless given this is intended for backends and the package is just type definitions anyway
1. We rely on the consumer to pass in the context type, meaning the generic signature becomes `<Event, Context, Output>`

Reply:
Ah okay, I might opt for option 1 just for ease of migration 🙇